tests/test_machine.py

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+
+import mkosi.machine as machine
+import pytest
+
+
+@pytest.mark.parametrize("boot", [["-d", "fedora", "-f", "--autologin", "--bootable", "--qemu-headless", "--hostonly-initrd", "qemu"], \
+                                    ["-d", "fedora", "-f", "--autologin", "boot"], \
+                                    ["-d", "fedora", "-f", "--autologin", "shell"]])
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")
+def test_run(boot):
+    m = machine.Machine(boot)
+    with m:
```

Comment:
Python doesn't have block scoped variables, so you can actually write this as:

```
with machine.Machine() as m:
```

and m will still be valid after the with block is finished